openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -4282,6 +4289,842 @@ def create_force(self, system, topology, **kwargs):
         system.addForce(gbsa_force)
 
 
+def validate_virtual_site_type_option(vsite_type):
```

Comment:
After our discussion, if this doesn't fit into the `ParameterAttribute.converter` framework., I think it's OK if these the methods remain where they are, though you should make them private (`_validate_virtual...`).

Reply:
Got converters working!